packages/nimble-components/src/wafer-map/modules/hover-handler.ts

```diff
@@ -52,7 +52,9 @@ export class HoverHandler {
             return;
         }
 
-        this.wafermap.hoverDie = this.wafermap.dataManager.getWaferMapDie(dieCoordinates);
+        if (this.wafermap.dataManager instanceof DataManager) {
```

Comment:
I think this check should be similar to the one below, i.e. `!this.wafermap.isExperimentalRenderer()
            && this.wafermap.dataManager instanceof DataManager`

Instead of duplicating that everywhere we can add a method to the wafer map?

```ts
public isLegacyDataManager() {
    return !this.wafermap.isExperimentalRenderer()
            && this.wafermap.dataManager instanceof DataManager;
}
```

Reply:
it seems typescript does not see the type constraints if I'm using a function, should I try and make a separate update function and get rid of this switching mechanism?